lib/ddtrace/contrib/sinatra/tracer_middleware.rb

```diff
@@ -19,23 +19,23 @@ def call(env)
             tracer.provider.context = context if context.trace_id
           end
 
-          # Begin the trace
-          tracer.trace(
-            Ext::SPAN_REQUEST,
-            service: configuration[:service_name],
-            span_type: Datadog::Ext::HTTP::TYPE_INBOUND
-          ) do |span|
-            Sinatra::Env.set_datadog_span(env, span)
+          Sinatra::Env.set_middleware_start_time(env)
 
+          # Run application stack
+          status, headers, response_body = @app.call(env)
+          [status, headers, response_body]
```

Comment:
I think I vaguely remember there was some change in the newer versions of Rack that changed what an acceptable return value was? (I think you were a part of that bug triage too.)

Is this legit? Just want to make sure there isn't any incompatibility here.

Reply:
I checked and the changes to Rack relate to the `response_body`, which we are just propagating here ([our changes related to the recent Rack change](https://github.com/DataDog/dd-trace-rb/pull/952/files)).
We only need headers here, so I'll change this a bit, but we are safe regardless.

Reply:
Code changes have been pushed. We are no longer touching the response.